drop testify dependency

kruskall opened the pull request:
makezero is only using `assert.ElementsMatch` once which can be easily replaced with `reflect.DeepEqual`

ashanbrown commented:
Seems reasonable given how little we use testify (as much as I do like it).  I created https://github.com/kruskall/makezero/pull/1 to update your branch with master, to include some other changes I made.  Happy to merge this if that looks ok.

ashanbrown commented:
I am curious to know what problem including testify was causing for you.  testify was only used in a test, so I don't expect it to be vendored in recent versions of go, for example.  Was it causing some sort of version conflict?

kruskall commented:
@ashanbrown I've rebased on the main branch and fixed the merge conflicts. You should be able to merge this now.

> I am curious to know what problem including testify was causing for you. testify was only used in a test, so I don't expect it to be vendored in recent versions of go, for example. Was it causing some sort of version conflict?

https://github.com/golang/go/issues/41431
